### DISCUSSION TITLE
Reference Schema In Separate Schema File

### DISCUSSION THREAD

**Discussion by valliappanmvi** [victools/jsonschema-generator#450] [Q&A] [answered]
I am using maven-plugin to generate schema for my java classes. For the nested classes, i see the reference schema generated is in the same file. 
Can we generate the reference schema in a separate file ?

**Comment by CarstenWickner** [comment, marked as answer]
Hi @valliappanmvi,

Short answer: Yes.

A similar question was raised on (that admittedly fairly long exchange on) #319, resulting in some examples being created, the most relevant for you should be the [`ExternalRefPackageExample`](https://github.com/victools/jsonschema-generator/blob/main/jsonschema-examples/src/main/java/com/github/victools/jsonschema/examples/ExternalRefPackageExample.java), where the schema generator is being configured to replace all types from a particular package with an external reference to a separate schema file.

Such a configuration could then be wrapped into a custom `Module` and included in your Maven plugin setup, where at the same time you'd instruct the plugin to generate those schema files for all classes in that same package.

That is how others are using it already, e.g., according to the question raised on #393.
